components/raftstore/src/store/worker/split_controller.rs

```diff
@@ -606,26 +607,50 @@ impl AutoSplitController {
             max_grpc_thread_count,
             max_unified_read_pool_thread_count,
             unified_read_pool_scale_receiver,
+            grpc_thread_usage_vec: vec![],
         }
     }
 
     pub fn default() -> AutoSplitController {
         AutoSplitController::new(SplitConfigManager::default(), 0, 0, None)
     }
 
+    fn update_grpc_thread_usage(&mut self, grpc_thread_usage: f64) {
+        self.grpc_thread_usage_vec.push(grpc_thread_usage);
+        let length = self.grpc_thread_usage_vec.len();
+        let detect_times = self.cfg.detect_times as usize;
+        // Only keep the last `self.cfg.detect_times` elements.
+        if length > detect_times {
+            self.grpc_thread_usage_vec.drain(..length - detect_times);
+        }
+    }
+
+    fn get_avg_grpc_thread_usage(&self) -> f64 {
+        let length = self.grpc_thread_usage_vec.len();
+        if length == 0 {
+            return 0.0;
+        }
+        let sum = self.grpc_thread_usage_vec.iter().sum::<f64>();
+        sum / length as f64
+    }
+
     fn should_check_region_cpu(&self) -> bool {
         self.cfg.region_cpu_overload_threshold_ratio > 0.0
     }
 
-    fn is_grpc_poll_busy(&self, grpc_thread_usage: f64) -> bool {
+    fn is_grpc_poll_busy(&self) -> bool {
         #[cfg(feature = "failpoints")]
         fail_point!("mock_grpc_poll_is_not_busy", |_| { false });
         if self.max_grpc_thread_count == 0 {
             return false;
         }
+        if self.cfg.grpc_thread_cpu_overload_threshold_ratio <= 0.0 {
+            return true;
+        }
+        let avg_grpc_thread_usage = self.get_avg_grpc_thread_usage();
         let grpc_thread_cpu_overload_threshold =
             self.max_grpc_thread_count as f64 * self.cfg.grpc_thread_cpu_overload_threshold_ratio;
-        grpc_thread_usage > 0.0 && grpc_thread_usage >= grpc_thread_cpu_overload_threshold
+        avg_grpc_thread_usage > 0.0 && avg_grpc_thread_usage >= grpc_thread_cpu_overload_threshold
```

Comment:
It seems that `grpc_thread_cpu_overload_threshold` should always positive? 
```suggestion
        avg_grpc_thread_usage >= grpc_thread_cpu_overload_threshold
```

Reply:
Since it's calculated by `self.max_grpc_thread_count as f64 * self.cfg.grpc_thread_cpu_overload_threshold_ratio`, it could be zero when `self.max_grpc_thread_count` is zero or `self.cfg.grpc_thread_cpu_overload_threshold_ratio` is too small. I think we better make sure the coner case is fully covered here.

Reply:
It seems they have been checked on the previous line L643 and L646?